src/components/ModalWithForm.js

```diff
@@ -23,6 +23,14 @@ export default class ModalWithForm extends Modal {
     });
   }
 
+  setLoading(isLoading, loadingText = "Saving...") {
+    if (isLoading) {
+      this._submitButton.textContent = loadingText;
+    } else {
+      this._submitButton.textContent = "Save";
+    }
+  }
+
```

Comment:
Needs correction (line 39): Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.

Reply:
Still needs correction